src/game/PhaserGame.tsx

```diff
@@ -0,0 +1,86 @@
+import { forwardRef, useEffect, useLayoutEffect, useRef } from 'react';
+import StartGame from './main';
+import { EventBus } from './EventBus';
+
+export interface IRefPhaserGame
+{
+    game: Phaser.Game | null;
+    scene: Phaser.Scene | null;
+}
+
+interface IProps
+{
+    currentActiveScene?: (scene_instance: Phaser.Scene) => void
+}
+
+export const PhaserGame = forwardRef<IRefPhaserGame, IProps>(function PhaserGame({ currentActiveScene }, ref)
+{
+    const game = useRef<Phaser.Game | null>(null!);
+
+    useLayoutEffect(() =>
+    {
+        if (game.current === null)
+        {
+
+            game.current = StartGame("game-container");
+
+            if (typeof ref === 'function')
+            {
+                ref({ game: game.current, scene: null });
+            } else if (ref)
+            {
+                ref.current = { game: game.current, scene: null };
+            }
+
+        }
+
+        return () =>
+        {
+            if (game.current)
+            {
+                game.current.destroy(true);
+                if (game.current !== null)
```

Comment:
**issue**: The check for `game.current !== null` is redundant here since it's already confirmed that `game.current` is not null. Consider removing this check.